link.go

```diff
@@ -554,12 +563,117 @@ func (a *LinkStats64) unmarshalBinary(b []byte) error {
 	return nil
 }
 
+var (
+	// registeredDrivers is the global map of registered drivers
+	registeredDrivers = make(map[string]reflect.Type)
+	// registeredSlaveDrivers is the global map of registered slave drivers
+	registeredSlaveDrivers = make(map[string]reflect.Type)
+)
+
+// RegisterDriver registers a driver with the link service
+// This allows the driver to be used to encode/decode the link data
+func RegisterDriver(d LinkDriver) error {
+	if _, yes := d.(LinkSlaveDriver); yes {
+		if _, ok := registeredSlaveDrivers[d.Kind()]; ok {
+			return fmt.Errorf("driver %s already registered", d.Kind())
+		}
+		registeredSlaveDrivers[d.Kind()] = reflect.TypeOf(d).Elem()
```

Comment:
This could simple be a normal assignment. :`registeredSlaveDrivers[d.Kind()] = d`